bindings/pyroot_experimental/PyROOT/src/DeclareCallablePyz.cxx

```diff
@@ -0,0 +1,663 @@
+// passes through the actual callable.
+PyObject* GenericCallableImpl_call(PyObject * /*self*/, PyObject *args)
+{
+   // Parse arguments with Python callable and get attributes with types
+   if (!PyTuple_Check(args)) {
+      PyErr_SetString(PyExc_RuntimeError, "Failed to parse arguments: Invalid tuple.");
+      return NULL;
+   }
+
+   if (!(PyTuple_Size(args) == 2)) {
+      PyErr_SetString(PyExc_RuntimeError, "Failed to parse arguments: Expect exactly one argument (Python callable).");
+      return NULL;
+   }
+
+   if (!(PyTuple_Size(args) == 2)) {
+      PyErr_SetString(PyExc_RuntimeError, "Failed to parse arguments: Expect exactly one argument (Python callable).");
+      return NULL;
+   }
+
+   auto instance = PyTuple_GetItem(args, 0);
+   auto pyfunc = PyTuple_GetItem(args, 1);
+   Py_INCREF(pyfunc);
+
+   if (!PyObject_HasAttrString(instance, "input_types")) {
+      PyErr_SetString(PyExc_RuntimeError, "Failed to create C++ callable: No input_types attribute found.");
+      return NULL;
+   }
+
+   if (!PyObject_HasAttrString(instance, "return_type")) {
+      PyErr_SetString(PyExc_RuntimeError, "Failed to create C++ callable: No return_type attribute found.");
+      return NULL;
+   }
+
+   auto inputTypes = PyObject_GetAttrString(instance, "input_types");
+   auto returnType = PyObject_GetAttrString(instance, "return_type");
+
+   // Extract name of Python callable
+   PyObject* pyname;
+   if (PyObject_HasAttrString(instance, "name")) {
+      pyname = PyObject_GetAttrString(instance, "name");
+   } else {
+      if (!PyObject_HasAttrString(pyfunc, "__name__")) {
+         PyErr_SetString(PyExc_RuntimeError, "Failed to create C++ callable: Python callable does not have attribute __name__.");
+         return NULL;
+      }
+      pyname = PyObject_GetAttrString(pyfunc, "__name__");
+   }
+
+   auto name = CPyCppyy_PyUnicode_AsString(pyname);
+   Py_DECREF(pyname);
+
+   // Get C++ return type
+   if (!CPyCppyy_PyUnicode_Check(returnType)) {
+      PyErr_SetString(PyExc_RuntimeError, "Failed to create C++ callable: Return type argument cannot be interpreted as string.");
+      return NULL;
+   }
+   std::string returnTypeStr = CPyCppyy_PyUnicode_AsString(returnType);
+   Py_DECREF(returnType);
+   if (returnTypeStr.compare("") == 0) {
+      returnTypeStr = "void";
+   }
+
+   // Put function in GenericCallable namespace
+   std::stringstream code;
+   code << "namespace ROOT {\n";
+
+   // Set return type
+   code << returnTypeStr << " ";
+
+   // Set name of Python callable as function name
+   code << name;
+
+   // Build function signature, type string and list of variables
+   code << "(";
+
+   auto iter = PyObject_GetIter(inputTypes);
+   auto inputTypesSize = PyObject_Size(inputTypes);
+   Py_DECREF(inputTypes);
+   if (!iter) {
+      PyErr_SetString(PyExc_RuntimeError, "Failed to create C++ callable: Failed to iterate over input types.");
+      return NULL;
+   }
+
+   std::map<std::string, std::string> typemap = {
```

Comment:
Can you comment on what this map contains? Map from C++ type to Python C API type name?

Reply:
Seems that they are only internal names you assign to types?